### PR TITLE
ci: add arm64 unit and integration lanes

### LINKED ISSUE #1009: CI: add an arm64 unit-test lane
No job runs on arm64 — every `runs-on` in `.github/workflows/*.yml` is `ubuntu-latest` and there is no `GOARCH` anywhere. `pierrec/lz4` ships hand-written per-architecture decode assembly, so the arm64 decoder is never executed by any gate.

The last two bumps of that dependency changed exactly that code: v4.1.27 (arm64 decode rework) and v4.1.29 (#1003 — `internal/lz4block/decode_arm64.s` +160/-43), which is the path behind every `LZ4Compressor.Decode` and `AppendDecompressed` call. A decode regression would reach Graviton / Apple-silicon users as corrupted CQL frames with nothing in CI to catch it.

Not hypothetical: upstream `00fc290` ("Fix LZ4 test fails on arm64", CASSGO-128) was a real arm64 LZ4 failure, fixed by bumping to v4.1.27 **and** adding a `unit-tests-arm64` job on `ubuntu-24.04-arm` wired into the downstream `needs:`. That commit isn't in this fork's master, so the lane is simply absent here.

Fix: port that job running `make test-unit`, which already reaches the nested `lz4` module (`Makefile:314`).

### PR DESCRIPTION
**Pull request by nikagra**
No job ran on arm64 — every `runs-on` in `.github/workflows/*.yml` was `ubuntu-latest` and there was no `GOARCH` anywhere in the repo. `pierrec/lz4` ships hand-written per-architecture decode assembly, so `internal/lz4block/decode_arm64.s` — the code behind every `LZ4Compressor.Decode` and `AppendDecompressed` call, and therefore behind every compressed native-protocol-v5 segment — was never executed by any gate.

Not hypothetical: `master` is already on `pierrec/lz4 v4.1.29` (#1003), whose diff is `internal/lz4block/decode_arm64.s +160/-43`, and the bump before that (v4.1.27) was itself an arm64 decode rework. Upstream hit a real arm64 LZ4 failure (CASSGO-128) and fixed it in `00fc290` by bumping to v4.1.27 **and** adding a `unit-tests-arm64` job on `ubuntu-24.04-arm` wired into the downstream `needs:`. That commit isn't in this fork's master, so the lane was simply absent here.

## What this does

- `build-arm64` on `ubuntu-24.04-arm` running `make test-unit`, with both integration jobs gated on it.
- `test-integration-scylla-arm64` — a ccm ScyllaDB `LATEST` cluster on `ubuntu-24.04-arm` with the integration suite run against it (added in review).
- The two Makefile assumptions that stood in the way of that second lane.

### The unit lane

- **Only `make test-unit`, not `make check`.** `check` is `golangci-lint` plus `go mod tidy -diff`, both architecture-independent; running it again would only re-download an arm64 golangci-lint to repeat the amd64 result. `test-unit` already descends into the nested `lz4` module (`go test -C lz4 -tags unit -race`), so the assembly above needs no separate step.
- **No `actions/cache` step.** The other jobs' key is `pr-check-${{ runner.os }}-${{ hashFiles('Makefile') }}`, and `runner.os` is `Linux` on both architectures, so sharing it would restore amd64 binaries into `bin/`. Nothing this job needs is expensive to recreate: `setup-go` caches modules under its own arch-scoped key, and `test-unit`'s `.prepare-pki` prerequisite regenerates the certs in seconds.
- **`build` is not renamed.** Upstream's commit also renames `build` → `unit-tests-amd64`; porting that would invalidate the existing `Build` status-check name for no benefit, since none of this fork's other job names match upstream's anyway.

### The integration lane

`Build (arm64)` shows the unit tests pass where the lz4 assembly lives, but nothing showed the driver actually talking to a cluster there — frame codec, shard-aware routing, TLS and the pooling paths were all still amd64-only. The new job starts a 3-node ccm ScyllaDB `LATEST` cluster on `ubuntu-24.04-arm` and runs `make test-integration-scylla` against it.

- **One version, one pass.** This lane is here to show the driver works end to end on aarch64, not to re-cover every ScyllaDB release, so the version matrix and the second `ccm`-tagged run stay on the amd64 job.
- **A separate job, not a matrix axis.** GitHub appends matrix values to a job's `name:`, so an `arch` axis would rename every existing `Integration Tests On Scylla (…)` status check — the same reason `build` isn't renamed above.
- **Both cache keys carry `runner.arch`.** `runner.os` is `Linux` on either architecture, so reusing the amd64 keys would hand this job an amd64 `get-version` in `bin/` and an x86_64 ScyllaDB relocatable in `~/.ccm`. The amd64 keys are left alone so their entries stay warm. `~/.sdkman` isn't cached here at all: `scylla-start` needs no JDK (ScyllaDB ≥ 6.0 uses the native `scylla nodetool`), and this image ships no Java 11 for the Makefile's `JAVA_HOME_11_X64` to resolve to.
- **The suite's default compressor.** The lz4 compressor lives in a nested module the root test harness cannot import — `createCluster` in `common_test.go` knows only `snappy` and `no-compression` — so this lane does not exercise the arm64 decode assembly; `Build (arm64)` still does that alone. Once the harness can build with lz4 (#1023), `TEST_COMPRESSOR=lz4` belongs on this job.

### Makefile

Two spots assumed an x86_64 host, and both sit on the path the new lane has to take:

- `.prepare-get-version` downloaded the `linux_amd64v3` release archive unconditionally. That binary is an `Exec format error` on aarch64, and both `resolve-*-version` targets depend on it, so version resolution would fail before any cluster is created. get-version publishes a `linux_arm64` archive of the same release; pick it from `uname -m`.
- scylla-ccm takes the relocatable package's architecture from `SCYLLA_ARCH` and falls back to `x86_64` (`ccmlib/scylla_repository.py`), so on aarch64 it would unpack a package whose `scylla` binary cannot run. `uname -m` already prints the two names ccm recognises.

Nothing changes on x86_64: `uname -m` selects the same `amd64v3` archive as before and matches ccm's own default. Editing the Makefile does change `hashFiles('Makefile')`, so the `pr-check-*` caches are cold once on this run; the `ccm-*` image caches are keyed on the resolved version and stay warm.

## Checks

`make check` and `make test-unit` (root and `lz4` modules) pass locally, the workflow parses, and every pinned action SHA in the new job already appears in the file. The rest is runtime behaviour that only CI can show — see `Build (arm64)` and `Integration Tests On Scylla (arm64)` on this run.

Fixes #1009
Refs: https://scylladb.atlassian.net/browse/DRIVER-961

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -57,10 +57,47 @@ jobs:
...
+      - uses: actions/setup-go@b7ad1dad31e06c5925ef5d2fc7ad053ef454303e # v7
+        with:
+          go-version-file: go.mod
+          cache-dependency-path: |
+            lz4/go.sum
+            tests/bench/go.sum
+            go.sum
+
+      - name: Run unit tests
+        run: make test-unit
```
[the remark this answers is not in the document]
Comment (dkropachev) [reply .github/workflows/main.yml:97]: +1

**Review by dkropachev — CHANGES_REQUESTED** [review]
@nikagra , can you please add one integration test on `LATEST` scylla for it too

File: .github/workflows/main.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (nikagra) [reply .github/workflows/main.yml:97]: Added `permissions: contents: read` to `build-arm64`, and to the new arm64 integration job below it. Folded into the commit that introduces the job.

**Comment by nikagra** [comment]
@dkropachev added `Integration Tests On Scylla (arm64)`: a ccm ScyllaDB `LATEST` cluster on `ubuntu-24.04-arm` with `make test-integration-scylla` run against it.

It needed two Makefile arch fixes to get there — `.prepare-get-version` was pinned to the amd64 archive (an `Exec format error` on aarch64, and both `resolve-*-version` targets depend on it), and scylla-ccm falls back to `SCYLLA_ARCH=x86_64`, so the runner would have unpacked a package whose `scylla` binary cannot run.

**Review by dkropachev — APPROVED** [review]
